### ISSUE TITLE
`expiring-todo-comments`: add regex option

### ISSUE THREAD

**Issue by Moeriki** [sindresorhus/eslint-plugin-unicorn#502]
I would love to be able to allow certain regexes.

### Use Case

I want to allow TODOs that reference a JIRA or GitHub issue id.

Eg.

```js
// TODO [JIRA-ISSUE-123] fix this
// TODO [#999] fix that
```

Could this be something useful to be included?

**Comment by sindresorhus** [comment]
Can you elaborate more on how the regex option would work? Maybe illustrate it with an example of how it’s config would look like.

**Comment by Moeriki** [comment]
### Config

```js
	{
		type: 'object',
		properties: {
			// …
			allowArguments: {
				type: 'array',
				item: { type: 'string' }
			},
		},
		additionalProperties: false
	}
```

### Example

```js
	{
		allowArguments: ["JIRA-CODE-\\d+"],
		allowWarningComments: false,
	}
```

### Code

The code should be [extended here](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/rules/expiring-todo-comments.js#L131) to iterate over the `allowArguments`.

### Notes

* The docs should make clear they run after the built-in matchers. Because in case of overlapping matches the `allowArguments` will not run.
* It makes no sense to check the `allowArguments` when `allowWarningComments` is set to `true`. We could check and warn on.

**Comment by Moeriki** [comment]
If there is interest I could prepare a PR.

**Comment by sindresorhus** [comment]
So you want to make `expiring-todo-comments` ignore the comment altogether? If so, I think just an `ignore` option that accepts an array of regexes that are matched on everything after `TODO ` would be more flexible.

**Comment by Moeriki** [comment]
That certainly would be more flexible.

So I would set the config to this.

```js
	{
		allowWarningComments: false,
		ignore: ["\bJIRA-CODE-\\d+\b"],
	}
```

So neither of these will give a warning.

```js
// TODO [JIRA-CODE-123] do this later
// TODO fix in JIRA-CODE-123
```

**Comment by sindresorhus** [comment]
Correct. PR welcome for that.

**Comment by Moeriki** [comment]
Question. Given this config.

```
	{
		allowWarningComments: false,
		ignore: ["\bJIRA-CODE-\\d+\b"],
	}
```

Should this be a warning?

```
// TODO [2020-01-01, JIRA-CODE-123] do this later
```

**Comment by sindresorhus** [comment]
No, `ignore` would *ignore* the comment altogether if matched.
